traefikee/tests/ingressclass-config_test.yaml

```diff
@@ -0,0 +1,100 @@
+suite: IngressClass configuration
+templates:
+  - ingressclass.yaml
+tests:
+  - it: should not provide an IngressClass when disabled
+    set:
+      ingressClass:
+        enabled: false
+    asserts:
+      - hasDocuments:
+          count: 0
+  - it: should provide a default IngressClass by default and prefer v1 api
+    set:
+      ingressClass:
+        enabled: true
+        isDefaultClass: true
+    asserts:
+      - isKind:
+          of: IngressClass
+      - isAPIVersion:
+          of: networking.k8s.io/v1
+      - equal:
+          path: metadata.annotations["ingressclass.kubernetes.io/is-default-class"]
+          value: "true"
+  - it: should be possible to provide a non-default IngressClass
+    set:
+      ingressClass:
+        enabled: true
+        isDefaultClass: false
+    asserts:
+      - isKind:
+          of: IngressClass
+      - isAPIVersion:
+          of: networking.k8s.io/v1
+      - equal:
+          path: metadata.annotations["ingressclass.kubernetes.io/is-default-class"]
+          value: "false"
+  - it: should provide an IngressClass with only v1beta1 api on Kubernetes v1.18
+    capabilities:
+      majorVersion: 1
+      minorVersion: 18
+    set:
+      ingressClass:
+        enabled: true
+        isDefaultClass: true
+    asserts:
+      - isKind:
+          of: IngressClass
+      - isAPIVersion:
+          of: networking.k8s.io/v1beta1
+  - it: should workaround and provide an IngressClass with v1 with only Capability (#698)
+    capabilities:
+      majorVersion: 1
+      minorVersion: 21
+      patchVersion: 14-eks-6d3986b
+    set:
+      ingressClass:
+        enabled: true
+        isDefaultClass: true
+    asserts:
+      - isKind:
+          of: IngressClass
+      - isAPIVersion:
+          of: networking.k8s.io/v1
+  - it: should fail when asking to deploy IngressClass on Traefik Enterprise < 2.3
+    set:
+      ingressClass:
+        enabled: true
+    chart:
+      appVersion: v2.1.0
+    asserts:
+      - failedTemplate:
+          errorMessage: "ERROR: IngressClass cannot be used with Traefik Enterprise < 2.3.0"
+  - it: should fail when asking to deploy IngressClass on Kubernetes < v1.16
+    set:
+      ingressClass:
+        enabled: true
+    capabilities:
+      majorVersion: 1
+      minorVersion: 15
+    asserts:
+      - failedTemplate:
+          errorMessage: "ERROR: You must use at least Kubernetes v1.16 with this Chart"
+  - it: should be possible to provide an IngressClass name
+    capabilities:
+      majorVersion: 1
+      minorVersion: 19
+    set:
+      ingressClass:
+        enabled: true
+        isDefaultClass: true
+        name: nondefaultname
+    asserts:
+      - isKind:
+          of: IngressClass
+      - isAPIVersion:
+          of: networking.k8s.io/v1
+      - equal:
+          path: metadata.name
+          value: nondefaultname
```

Comment:
I would add tests for the non default image tag in `imageVersion` helper using `image.tag`.

Reply:
@darkweaver87 I did add a test for the "-" image tags.
Are you referring to a Helm test with special custom image if user rename it without semVer ?

Reply:
No just the one we are providing ;-)